src/hotspot/share/oops/oop.cpp

```diff
@@ -154,7 +154,8 @@ bool oopDesc::is_typeArray_noinline()   const { return is_typeArray();   }
 
 bool oopDesc::has_klass_gap() {
   // Only has a klass gap when compressed class pointers are used.
-  return UseCompressedClassPointers;
+  // Except when using compact headers.
```

Comment:
```suggestion
  // Only has a klass gap when compressed class pointers are used and not
  // using compact headers.
```
(Not sure if repeating the fairly simple disjunction below makes sense, but there has been a comment before too)